cfgmgr/nbrmgr.cpp

```diff
@@ -257,3 +386,20 @@ void NbrMgr::doTask(Consumer &consumer)
         it = consumer.m_toSync.erase(it);
     }
 }
+
+void NbrMgr::doTask(Consumer &consumer)
+{
+    string table_name = consumer.getTableName();
+
+    if (table_name == CFG_NEIGH_TABLE_NAME)
+    {
+        doSetNeighTask(consumer);
+    }
+
+    if (table_name == APP_NEIGH_RESOLVE_TABLE_NAME)
+    {
+        doDeleteNeighTask(consumer);
```

Comment:
As we discussed, can this be a generic Resolve neighbor API?

Reply:
Done. But note that I am still keeping doResolveNeighTask() separate from doSetNeighTask(). There are enough differences between the functions to add 3+ if else statements. Hence decided to keep them separate. Also as both of them process information from different APP DB tables, good to keep them separate keeping their evolution in mind. But yes, they both end up calling setNeighbor() as you suggested! Thanks for your help in making the solution brief and generic!